src/js/other-websites/icon.js

```diff
@@ -0,0 +1,452 @@
+  iframe.style.top = `${elementBounds.bottom}px`;
+  if (elementBounds.right < RELAY_INPAGE_MENU_WIDTH) {
+    // If the right-hand side of the input is closer to the left-hand side of
+    // the page than the menu is wide, align the left-hand side of the menu with
+    // the left-hand side of the input:
+    iframe.style.left = `${elementBounds.left}px`;
+    iframe.style.right = "";
+  } else {
+    // But otherwise, we can just align the right-hand side of the menu with the
+    // right-hand side of the input, to keep it close to the button:
+    iframe.style.left = "";
+    iframe.style.right = `calc(100vw - ${elementBounds.right}px)`;
+  }
+
+  const iframeHeight = Number.parseInt(
+    getComputedStyle(iframe).height.replace("px", ""),
+    10
+  );
+  if (Number.isNaN(iframeHeight)) {
+    iframe.style.top = `${elementBounds.bottom}px`;
+  } else {
+    const elementBounds = target.getBoundingClientRect();
+    const viewportHeight = document.documentElement.clientHeight;
+    const iframeSrc = new URL(iframe.getAttribute("src"));
+    if (
+      viewportHeight - elementBounds.bottom < iframeHeight &&
+      elementBounds.top > iframeHeight
+    ) {
+      // Position the menu at the top of the input element it there is room there,
+      // and no room below it:
+      iframeSrc.hash = "#position=top";
+      iframe.setAttribute("src", iframeSrc.href);
+      iframe.style.top = "unset";
+      iframe.style.bottom = `${viewportHeight - elementBounds.top}px`;
+    } else {
+      iframeSrc.hash = "#position=bottom";
+      iframe.setAttribute("src", iframeSrc.href);
+      iframe.style.top = `${elementBounds.bottom}px`;
+      iframe.style.bottom = "unset";
+    }
+  }
+}
+
+/**
+ * @param {MouseEvent} event 
+ * @returns void
+ */
+function onEmailInputHover(event) {
+  const element = event.target;
+  if (!(element instanceof HTMLInputElement)) {
+    return;
+  }
+
+  // Apply the above styles only when the cursor hovers the Relay icon:
+  if (intersectsRelayIcon(element, { x: event.offsetX })) {
+    applyIconHoverStyle(element);
+  } else {
+    applyIconUnhoverStyle(element);
+  }
+}
+
+/**
+ * Whenever the Relay icon is hovered, we add class styles for the <input>
+ * element that icon is in that activate on hover or blur.
+ *
+ * Since these styles are specific to that element (because they preserve its
+ * existing styles), this function has to be called on every hover — otherwise,
+ * the class might have its styles defined for a different <input>.
+ *
+ * @param {HTMLInputElement} element
+ */
+function setHoverStyles(element) {
+  // Styles to add hover effects (background glow and a pointer cursor),
+  // while preserving the existing background styles for this element:
+  const elementStyles = getComputedStyle(element);
+  const hoverBackgroundStyle =
+    elementStyles.backgroundImage +
+    `, linear-gradient(to left, #f68fff ${
+      getPaddingRight(element) * 2 + 25
```

Comment:
Suggestion: Can you move `#f68fff` to a variable at the top of the file? I feel like this is hidden/buried CSS in this file that we'd want to better surface.

Reply:
Extra suggestion: Move all/any CSS attributes to the top of the file.

Reply:
Moved all colors in 3722328. Not sure if there were additional properties you were thinking of? (Note that most styles are dynamically generated based on existing styles for a particular input element, so I can't just move them to the top as hardcoded strings.)